src/addrdb.h

```diff
@@ -58,12 +58,10 @@ std::optional<bilingual_str> LoadAddrman(const std::vector<bool>& asmap, const A
  */
 void DumpAnchors(const fs::path& anchors_db_path, const std::vector<CAddress>& anchors);
 
-/**
- * Read the anchor IP address database (anchors.dat)
- *
- * Deleting anchors.dat is intentional as it avoids renewed peering to anchors after
- * an unclean shutdown and thus potential exploitation of the anchor peer policy.
- */
+/** Read the anchor IP address database (anchors.dat) */
```

Comment:
This PR looks to improve things based on the concept however I could not understand why is the file deleted? Why not just read and write? 

1. Dump block-relay-only peers to anchors.dat on exit
2. Read anchors.dat on start
3. Repeat 1

Then I read the comment which is removed by the PR:

_Deleting anchors.dat is intentional as it avoids renewed peering to anchors after an unclean shutdown and thus potential exploitation of the anchor peer policy_

So maybe this comment should not be removed or add relevant information.

```suggestion
/**
* Read the anchor IP address database (anchors.dat)
*
* Deleting anchors.dat after trying to connect to block-relay-only peers is intentional as...
 */
 ```

Reply:
Just read and write?  Why keep in anchors.dat addresses we could not be able to connect, for example?
Supposing we are on your step 2, what would happen if that peers were offline?

Reply:
Replace with new peers?

Reply:
I understood the point. I removed it after https://github.com/bitcoin/bitcoin/pull/24034#discussion_r782789428, did you check that? what do you think?